contracts/src/periphery/EigenDACertVerifierV2.sol

```diff
@@ -0,0 +1,260 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.9;
+
+import {IEigenDACertVerifierV2} from "src/periphery/interfaces/IEigenDACertVerifierV2.sol";
+import {IEigenDAThresholdRegistry} from "src/interfaces/IEigenDAThresholdRegistry.sol";
+import {IEigenDASignatureVerifier} from "src/interfaces/IEigenDASignatureVerifier.sol";
+import {IEigenDARelayRegistry} from "src/interfaces/IEigenDARelayRegistry.sol";
+import {EigenDACertVerificationV2Lib as CertLib} from "src/periphery/libraries/EigenDACertVerificationV2Lib.sol";
+import {OperatorStateRetriever} from "lib/eigenlayer-middleware/src/OperatorStateRetriever.sol";
+import {IRegistryCoordinator} from "lib/eigenlayer-middleware/src/interfaces/IRegistryCoordinator.sol";
+import "src/interfaces/IEigenDAStructs.sol";
+
+/**
+ * @title EigenDACertVerifierV2 - EigenDA V2 certificate verification
+ * @author Layr Labs, Inc.
+ * @notice Contains all V2-specific verification functionality
+ */
+contract EigenDACertVerifierV2 is IEigenDACertVerifierV2 {
+    error InvalidSecurityThresholds();
+
+    /// @notice The EigenDAThresholdRegistry contract address
+    IEigenDAThresholdRegistry public immutable eigenDAThresholdRegistryV2;
+
+    /// @notice The EigenDASignatureVerifier contract address
+    IEigenDASignatureVerifier public immutable eigenDASignatureVerifierV2;
+
+    /// @notice The EigenDARelayRegistry contract address
+    IEigenDARelayRegistry public immutable eigenDARelayRegistryV2;
+
+    /// @notice The EigenDA middleware OperatorStateRetriever contract address
+    OperatorStateRetriever public immutable operatorStateRetrieverV2;
+
+    /// @notice The EigenDA middleware RegistryCoordinator contract address
+    IRegistryCoordinator public immutable registryCoordinatorV2;
+
+    /// @notice Security thresholds for V2 certificate verification
+    /// Storing individual fields as immutable instead of the struct
+    uint8 public immutable securityThresholdsConfirmation;
+    uint8 public immutable securityThresholdsAdversary;
```

Comment:
@samlaf #1472 

prob good to add small description of what these are and how they are used

Reply:
https://github.com/Layr-Labs/eigenda/pull/1477/commits/5f4c14a531d297dc936940c7b45b36bd20c6cc30